### ISSUE TITLE
multiple instances on the same page?

### ISSUE THREAD

**Issue by paul-shannon** [igvteam/igv.js#700]
I [see](https://github.com/igvteam/igv.js/issues/197) that a couple of years ago this was neither supported nor planned.   Any change since then?  I have a round draft of i[gvShiny](https://github.com/paul-shannon/igvShiny) for which this would be very useful.

Attempting this now (see minimal code below) I get ````Cannot read property 'indexOf' of undefined````.

````
<div id="igvDiv0" style="padding-top: 50px;padding-bottom: 20px; height: auto"></div>
<div id="igvDiv1" style="padding-top: 50px;padding-bottom: 20px; height: auto"></div>
<script type="text/javascript">

 $(document).ready(function () {
    var igvDiv, options;
    igvDiv0 = $("#igvDiv0")[0];
    igvDiv1 = $("#igvDiv1")[0];
    options = {locus: "MEF2C",
               minimumBases: 5,
               showRuler: true,
               reference: {id: "hg38",
                     fastaURL: "https://s3.amazonaws.com/igv.broadinstitute.org/genomes/seq/hg38/hg38.fa",
                  cytobandURL: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg38/cytoBandIdeo.txt"
                  },
               tracks: [
                 {name: 'Gencode v24',
                       url: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg38/genes/gencode.v24.annotation.sorted.gtf.gz",
                  indexURL: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg38/genes/gencode.v24.annotation.sorted.gtf.gz.tbi",
                  format: 'gtf',
                  visibilityWindow: 2000000,
                  displayMode: 'EXPANDED'
                  },
                 ]
              }; // options
        browser1 = igv.createBrowser(igvDiv0, options);
        browser2 = igv.createBrowser(igvDiv1, options);
    });

</script>
````

**Comment by jrobinso** [comment]
Its a desirable feature but there are many desirable features,   I'm not working on that one at the moment but have been chipping away at it.  Many items depend on a global reference genome and a few other properties of "browser",  these need refactored.   

Second point,  if you are running from the latest in master igv.createBrowser() now returns a promise which resolves when the browser is ready.   This is a 2.0 change,  not backward compatible.  The fix for application code is easy enough,  see below.   The wiki doc has been updated to reflect this

2.0 pattern:
```javascript
    igv.createBrowser(igvDiv0, options)
          .then(function (browser) {
                  browser1 = browser
          }      
    });
```

**Comment by paul-shannon** [comment]
Makes sense, Jim.   

Thanks for the pointer on the promise pattern - I now use that in igvShiny.

**Comment by jrobinso** [comment]
Opening this issue as a request.

**Comment by yarongolan** [comment]
Would be great to have multiple browser support. We currently have to move the browser between parents when displaying it in minimal and then expanded modes.

**Comment by jrobinso** [comment]
Yes, I'm working towards it.  The first step is to remove  165+ references to "igv.browser" to refactor,  I remove 20 or so this evening.     

@yarongolan Could you expand on what you mean by "minimal" and "expanded" modes?   If you have a public website maybe I can see it.

**Comment by jrobinso** [comment]
OK, thanks @yarongolan , its helpful to know what people are doing.   Multiple browsers will happen, probably in a 2.1 release in the fall.  Right now I have new features frozen for a 2.0 release, for which we are hopefully close.

**Comment by yarongolan** [comment]
Looking forward to v2.0 and thanks! - I'll keep an eye on this thread.

**Comment by jrobinso** [comment]
We're testing this now in the branch "multiple-browsers".   See examples/multipleBrowsers.html in that branch.

**Comment by jrobinso** [comment]
OK, this is done. We decided this was important enough to delay 2.0 a few days.     Documentation to follow, but see https://igv.org/web/test/examples/multipleBrowsers.html  for an example (https://github.com/igvteam/igv.js/blob/master/examples/multipleBrowsers.html)

**Comment by yarongolan** [comment]
Much appreciated!

**Comment by jrobinso** [comment]
@yarongolan Note the 2.0 pattern,  igv.createBrowser returns a promise,  not the "browser" object, the example should make that clear.

**Comment by yarongolan** [comment]
@jrobinso Sorry to bother, but one issue currently is that elements have the same IDs when creating more than one browser.

**Comment by jrobinso** [comment]
Ok, that's a mistake, will fix.

**Comment by turner** [comment]
I'm on the duplicate id issue now.

**Comment by turner** [comment]
@yarongolan duplicate element ids should all be gone now. Fixed with commit f868d9e9
